commands.c

```diff
@@ -995,30 +997,29 @@ static void set_copy_flags(struct Email *e, bool decode, bool decrypt,
 /**
  * mutt_save_message_ctx - Save a message to a given mailbox
  * @param e                Email
- * @param delete_original  If true, delete the original
- * @param decode           If true, decode the message
- * @param decrypt          If true, decrypt the message
+ * @param save_opt         Copy or move (delete the original?)
+ * @param transform_opt    Decrypt / Decode / None
  * @param m                Mailbox to save to
  * @retval  0 Success
  * @retval -1 Error
  */
-int mutt_save_message_ctx(struct Email *e, bool delete_original, bool decode,
-                          bool decrypt, struct Mailbox *m)
+int mutt_save_message_ctx(struct Email *e, enum MessageSaveOpt save_opt,
+                          enum MessageTransformOpt transform_opt, struct Mailbox *m)
 {
   CopyMessageFlags cmflags = MUTT_CM_NO_FLAGS;
   CopyHeaderFlags chflags = CH_NO_FLAGS;
   int rc;
 
-  set_copy_flags(e, decode, decrypt, &cmflags, &chflags);
+  set_copy_flags(e, transform_opt, &cmflags, &chflags);
 
-  if (decode || decrypt)
+  if (transform_opt)
```

Comment:
Please check explicitely: the point of having an enum is it allows us to map meaning to symbols ("decode" -> TRANSFORM_DECODE). You don't want the "then" branch to be automatically hit the day you add a TRANSFORM_FOO.